### PR TITLE
Reset focus when hero was dismissed, use KEY_RETURN for EVENT_OPENFOCUS by default

### LINKED ISSUE #4138: Adventure map, [ENTER] / [SPACE] hotkeys behavior
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

According to the default hotkeys config, `ENTER` should open focused element, which can be a hero or a town:
```
# 2.3 game events:
...
# open focus		- (default keycode:  13 = 'return')
```

This is also how it works in the original game.

In fheroes2 pressing `ENTER` doesn't open either of the dialogs.  Pressing `SPACE` (to which the `default action` is bound by default) opens the currently selected town, but does not open the currently selected hero.

Both hotkeys should open town/hero depending on the current focus.

### Save file

No save file needed.

### Additional info

_No response_

### LINKED ISSUE #4178: Adventure map, hero scrollbar doesn't update to be centered, after some hero disappeared
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

While running fheroes2 at low resolution (up to 800x600), scrollbar doesn't become centered, in a case we dismiss a hero when all heroes doesn't fit available boxes in hero column.
Here we have 8 heroes for 7 visible slots:
image
We dismiss some heroes:
image
Scrollbar is still sticked to upper arrow. It cannot be moved.
But when we save and reload the game bar appears correctly:
image
Note, that when we loose hero in battle everything works correctly.

### Save file

[Test Scroll.zip](https://github.com/ihhub/fheroes2/files/7128550/Test.Scroll.zip)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #4138
fix #4178

### CODE REVIEW

**Comment by a1exsh** [comment]
I'm noticing an issue with the focus after closing the castle dialog.  Will provide details and a save file soon.

**Comment by a1exsh** [comment]
> I'm noticing an issue with the focus after closing the castle dialog. Will provide details and a save file soon.

Ah, actually all is fine: the focus jumps to the hero in the castle, if there is one, after closing the castle dialog.  I think that's helpful and is by design. :)

**Comment by oleg-derevenetz** [comment]
> Ah, actually all is fine: the focus jumps to the hero in the castle, if there is one, after closing the castle dialog.  I think that's helpful and is by design. :)

This PR doesn't touch the castle dialog logic. If no one has complained about this behavior yet, I believe it is correct.

**Comment by a1exsh** [comment]
> This PR doesn't touch the castle dialog logic. If no one has complained about this behavior yet, I believe it is correct.

Yes, I've compared with the HEAD and this part works the same way there. I haven't checked the behavior on dismiss, though.

File: src/fheroes2/game/game_startgame.cpp [hunk 1]
```diff
@@ -231,48 +231,46 @@ void Game::OpenHeroesDialog( Heroes & hero, bool updateFocus, bool windowIsGameW
...
-            if ( needFade )
-                needFade = false;
-
-            switch ( result ) {
-            case Dialog::PREV:
-                if ( it == myHeroes.begin() )
-                    it = myHeroes.end();
-                --it;
-                break;
+    int result = Dialog::ZERO;
```
Comment (ihhub) [inline src/fheroes2/game/game_startgame.cpp:234]: I think besides this change we have to modify `void SetCurrentVisible` method to move **UpdateScrollbarRange();** call from `if ( _currentId >= 0 && ( _topId > _currentId || _topId + maxItems <= _currentId ) ) { // out of view` condition. Just this fix solves the whole problem with scrollbar. What do you think?

Comment (oleg-derevenetz) [reply src/fheroes2/game/game_startgame.cpp:234]: @ihhub May be. `ResetFocus()` also calls `ListBox::Reset()` which properly sets scrollbar range. If you wish, you can add commit with this change to this PR, or I can do it tomorrow - I need to move out now :)

**Comment by ihhub** [comment]
@oleg-derevenetz , a bunch of thanks for fixing these issues!

**Pull request merged**